csp_billing_adapter_microsoft/plugin.py

```diff
@@ -81,55 +83,74 @@ def get_account_info(config: Config):
     The information contains the metadata for compute and network.
     """
     account_info = _get_metadata()
-    account_info['compute'] = json.loads(account_info.get('compute', '{}'))
-    account_info['network'] = json.loads(account_info.get('network', '{}'))
     account_info['cloud_provider'] = get_csp_name(config)
 
     return account_info
 
 
-def get_api_header():
-    """
-    Get the header to be used in requests
-
-    Prefer IMDS which requires a token.
-    """
-    request = urllib.request.Request(
-        METADATA_TOKEN_URL,
-        headers={'Metadata': 'True'},
-        method='GET'
+def _get_metadata():
+    """Return a dict containing compute, network and signature information."""
+    metadata = {}
+    try:
+        metadata = _get_instance_metadata()
+        metadata['attestedData'] = _get_signature()
+    except ValueError as error:
+        log.error('Could not load JSON from metadata: {}'.format(error))
+        for key in ['compute', 'network', 'attestedData']:
+            if key not in metadata:
+                metadata[key] = {}
+
+    return metadata
+
+
+def _get_instance_metadata():
+    """Return all compute and network information from metadata."""
+    api_version = _get_latest_api_version()
+    if not api_version:
+        # oldest API version
+        api_version = OLDEST_API_VERSION
```

Comment:
We are prepared for the API versions check to turn up empty, fair enough. But later we assume `'2020-09-01'` of he APi is available. If that assumption is valid then why check for the versions and not simply use `'2020-09-01'` throughout the code rather than having this self consistency issue?